presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/AssignmentLimiter.java

```diff
@@ -99,6 +102,11 @@ public void shutdown()
 
     public synchronized void checkAssignFrom(String nodeIdentifier)
     {
+        int currentNodeCount = nodeSupplier.getWorkerNodes().size();
+        if (currentNodeCount < minimumNodeCount) {
+            throw new PrestoException(RAPTOR_NOT_ENOUGH_NODES, format("Too many nodes down (required: %s, current: %s)", minimumNodeCount, currentNodeCount));
```

Comment:
Down is too specific - I don't think we can say that confidently. I would recommend 'Not enough nodes available (required: %s, current: %s)'.